Sources/Datadog/RUM/RUMVitals/VitalInfoSampler.swift

```diff
@@ -54,7 +52,7 @@ internal final class VitalInfoSampler {
         cpuReader: SamplingBasedVitalReader,
         memoryReader: SamplingBasedVitalReader,
         refreshRateReader: ContinuousVitalReader,
-        frequency: TimeInterval = VitalInfoSampler.frequency,
+        frequency: TimeInterval?,
```

Comment:
How about requiring `frequency` and having optional `VitalInfoSampler` in `RUMViewScope`? That could be more consistent IMO, as `nil` would mean that Mobile Vitals are disabled. Now, if passing `frequency: nil` we make the sampler no-op.

This is not blocking, I'm ok with either way 👍, just wondering what's your take on this @maxep

Reply:
The `refreshRateReader` which is of type `ContinuousVitalReader` is not impacted by the `frequency` parameter, so should we disable the refresh vital as well if the frequency is `.never`?

Reply:
Ah, it makes sense then. What I suggested would require more refactoring to have `nil` instead of no-ops (e.g. having multiple samplers, one for each `reader`). Definitely not sth worth doing now if no other reason 👍.

> so should we disable the refresh vital as well if the frequency is .never?

This is a very good question though 🤔💭. AFAIK on Android we disable Refresh Rate vitals if `.never` is provided (cc @xgouchet @mariusc83). Also, "disable" is what we state in the doc:
```swift
/// Don't provide mobile vitals.
case never
```

Reply:
So we confirmed to disable **all** vitals ✅